### PR TITLE
feat: [PL-39269]: load certificate from a common source of truth

### PR DESCRIPTION
**Pull request by xingchi-jin**
## Describe your changes
1. Certificate will be loaded from a common source of truth specified by env CA_CERTS_DIR
2. With many experiments, only x509 ca certs can be supported by both RHEL and keytool. PEM format can contain multiple certs in a bundle file, while DER format can contain only one
3. When it's a PEM file, it will reach blocks one by one and import to truststore. 
4. Updated the connectivity check script, because when test with SMP environment, I realize the SMP ca certs are EV certs, which doesn't exist in normal RHEL ca bundle file even after running update-ca-trust. It exists in ca.bundle.trust.crt
5. the load cert script will only be executed when CA_CERTS_DIR is set AND when there are files in CA_CERTS_DIR 
6. all the pem files in the CA_CERTS_DIR will be splited to individual pem files and put in SHARED_CA_CERTS_PATH folder

Tests
1. Tested with SMP 
image
image
2. Connectivity script test
image

3. In the cert sample folder, i have 3 cert pem files. openssl.pem and smp.pem both have two x509 cert blocks in it. I have set a env ADDITIONAL_CERTS_PATH (which is same as the current name CI is using). The script will split the x509 pem blocks into each individual file and put them in ADDITIONAL_CERTS_PATH
image
image
image

4. tested with old CI cert solution. no regression. I used helm to deploy, CI certs are set as below (second screenshot). The first screenshot shows the certs were mounted to LE pod
image
image

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51054)

### CODE REVIEW

**Comment by xingchi-jin** [comment]
trigger uts

**Comment by xingchi-jin** [comment]
trigger uts

File: dockerization/delegate/Dockerfile-minimal [hunk 1]
```diff
@@ -47,9 +49,10 @@ ENV CLIENT_TOOLS_DOWNLOAD_DISABLED=true
 ENV INSTALL_CLIENT_TOOLS_IN_BACKGROUND=true
 ENV JAVA_HOME=/opt/java/openjdk/
 ENV PATH="$JAVA_HOME/bin:${PATH}"
+ENV ADDITIONAL_CERTS_PATH=/opt/harness-delegate/additional_certs_pem_split
 
-RUN curl -s -L -o delegate.jar $BASEURL/$DELEGATEVERSION/delegate.jar
-
+#RUN curl -s -L -o delegate.jar $BASEURL/$DELEGATEVERSION/delegate.jar
+COPY delegate.jar /opt/harness-delegate/
```
Comment (bharatgoelharness) [inline dockerization/delegate/Dockerfile-minimal:55]: @xingchi-jin why you have changed this?

**Review by bharatgoelharness — CHANGES_REQUESTED** [review]

File: dockerization/delegate/Dockerfile-minimal (hunk 1 above)
Comment (xingchi-jin) [reply dockerization/delegate/Dockerfile-minimal]: sorry, removed

File: dockerization/delegate/Dockerfile [hunk 2]
```diff
@@ -70,6 +72,7 @@ ENV PATH=/opt/harness-delegate/client-tools/go-template/v0.4.4/:$PATH
 ENV PATH=/opt/harness-delegate/client-tools/chartmuseum/v0.15.0/:$PATH
 ENV PATH=/opt/harness-delegate/client-tools/tf-config-inspect/v1.3/:$PATH
 ENV PATH=/opt/harness-delegate/client-tools/kustomize/v4.5.4/:$PATH
+ENV ADDITIONAL_CERTS_PATH=/opt/harness-delegate/additional_certs_pem_split
```
[the remark this answers is not in the document]
Comment (xingchi-jin) [reply dockerization/delegate/Dockerfile:73]: the ADDITIONAL_CERTS_PATH is described in the CI document https://developer.harness.io/docs/continuous-integration/use-ci/set-up-build-infrastructure/k8s-build-infrastructure/configure-a-kubernetes-build-farm-to-use-self-signed-certificates/#enable-self-signed-certificates 

I think the HARNESS_ADDITIONAL_CERTS_DIR is specified on the CI lite-engine

[the remark this answers is not in the document]
Comment (williamwissemann) [reply dockerization/delegate/Dockerfile:73]: My understanding is that `ADDITIONAL_CERTS_PATH` & `CI_MOUNT_VOLUMES` (in the docs) are combined to from `HARNESS_ADDITIONAL_CERTS_DIR` and `HARNESS_ADDITIONAL_CERTS_LIST`. Without `CI_MOUNT_VOLUMES` being specified, I'm not sure the certificates will be passed through to the CI steps.
  
 CC: @r0610205

**Review by r0610205 — APPROVED** [review]

**Comment by xingchi-jin** [comment]
trigger uts

**Review by bharatgoelharness — DISMISSED** [review]

File: dockerization/delegate/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (xingchi-jin) [reply dockerization/delegate/Dockerfile]: we have addressed this in our meeting.

**Comment by xingchi-jin** [comment]
trigger uts

**Comment by xingchi-jin** [comment]
trigger uts

**Review by bharatgoelharness — APPROVED** [review]

**Review by raghuAtWings — APPROVED** [review]

**Review by xingchi-jin — COMMENTED** [review]
> Reviewed 1 of 3 files at r1, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51054)* status: 5 file(s) to review. 2. unresolved discussions.

**Pull request merged**